### PR TITLE
feat: implement logic for insert/remove products

### LINKED ISSUE #14: Define topic for product update

### LINKED ISSUE #15: Update Display info according to product update

### LINKED ISSUE #16: Update shelf internal state (product order)

### LINKED ISSUE #17: Study how to manage the persistence of the products

### LINKED ISSUE #18: Manage insert/remove product from the shelf

### PR DESCRIPTION
**Pull request by nicolasfara**
This PR introduce:
  - Logic for manage insert a new product in the shelf
  - Logic for manage remove a product from the shelf

Closes #14
Closes #15
Closes #16
Closes #17
Closes #18

### CODE REVIEW

File: devices/product_manager.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    @post_load
+    def make_user(self, data, **kwargs):
+        #pylint: disable=missing-function-docstring,unused-argument,no-self-use
+        return ProductShelf(**data)
+
+
+class ProductManager:
+    #pylint: disable=too-many-instance-attributes
+    """
+    TODO.
```
Comment (vitlinda) [inline devices/product_manager.py]: This TODO is still necessary ? Also at line 22, 27 and 66

File: devices/product_manager.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            self.__logger.debug("Now products %s", self.__products.products)
+            product_in_list = list(
+                filter(
+                    lambda p: p.code == readed_product.code and p.lot == readed_product.lot,
+                    self.__products.products
+                )
+            )
+
+            if not product_in_list:
+                self.__logger.debug("The product not exist, insert in the shelf")
```
Comment (vitlinda) [inline devices/product_manager.py]:
```suggestion
                self.__logger.debug("The product doesn't exist, insert in the shelf")
```

File: models/product.py [hunk 3]
```diff
@@ -1,29 +1,63 @@
...
-        return pickle.dumps(self)
-
-def create_product(product_id: str, name: str, price: str, expiration_day: str) -> Product:
-    """
-    Create a new Product object from parameters.
-    """
-    exp_date = datetime.strptime(expiration_day, "%d/%m/%Y")
-    return Product(product_id=product_id, name=name, price=price, expiration_day=exp_date)
+    #pylint: disable=too-few-public-methods
+    """TODO"""
```
Comment (vitlinda) [inline models/product.py]: As above

File: models/product.py [hunk 4]
```diff
@@ -1,29 +1,63 @@
 """
 TODO.
```
Comment (vitlinda) [inline models/product.py:2]: As above

**Review by vitlinda — APPROVED** [review]

File: devices/product_manager.py (hunk 1 above)
Comment (nicolasfara) [reply devices/product_manager.py]: Yes is the documentation of the class.

**Pull request merged**